### ISSUE TITLE
Fix section 508 accessibility warning problems

### ISSUE THREAD

**Issue by msecret** [18F/openFEC#584]
According to the accessibility tool I used, warning problems are severe and should be fixed.

### red
WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.InputText.Name (2 occurences)
   This text input element does not have a name available to an accessibility API. Valid names are: label element, title attribute.
```
<input type="text" name="search" class="search-bar tt-input" placeholder="Enter a candidate name" autocomplete="off" spellcheck="false" dir="auto" style="position: relative; vertical-align: top;">
```
```
<input id="glossary-search" type="text" placeholder="Search for terms" class="tt-input" autocomplete="off" spellcheck="false" dir="auto" style="position: relative; vertical-align: 
```

WCAG2AA.Principle1.Guideline1_3.1_3_1.F68 (2 occurences)
   This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
```
<input type="text" name="search" class="search-bar tt-input" placeholder="Enter a candidate name" autocomplete="off" spellcheck="false" dir="auto" style="position: relative; vertical-align: top;">
```
```
<input id="glossary-search" type="text" placeholder="Search for terms" class="tt-input" autocomplete="off" spellcheck="false" dir="auto" style="position: relative; vertical-align: top;">
```

WCAG2AA.Principle1.Guideline1_3.1_3_1.H42.2
   Heading tag found with no content. Text that is not intended as a heading should not be marked up with heading tags.
```
<h4><dt><span id="glossary-term" cl...</h4>
```

WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Button.Name (2 occurences)
   This button element does not have a name available to an accessibility API. Valid names are: title attribute, element content.
```
<button id="glossary-toggle" data-slide="right" class="header-nav__link side-toggle"><i class="ti ti-help-alt"></i></button>
```
```
<button class="button--close u-float-right" id="hide-glossary"><i class="ti-close text--white"...</button>
```

WCAG2AA.Principle3.Guideline3_2.3_2_2.H32.2 (2 occurences)
   Form does not contain a submit button (input type="submit", input type="image", or button type="submit").
```
<form id="search" action="/"> <span class="twitt...</form>
```
```
<form id="large-search" action="/"> <label for="search" ...</form>
```

WCAG2AA.Principle3.Guideline3_1.3_1_1.H57.2
   The html element should have a lang or xml:lang attribute which describes the language of the document.

**Comment by noahmanger** [comment]
Ok, this is taken care of. There's still a couple of red warnings because the forms are submitting with a `<button>` instead of `<input type="submit">`. We think that's ok, but will revisit with the full 508 check later.
